manager.java

```diff
@@ -1,70 +1,51 @@
-        if (totalSeats < 60) {
-            int profit = totalSeats * 10;
-            String printProfit = "$" + String.valueOf(profit);
-            System.out.println(printProfit);
-        } else {
-            if (numOfRows % 2 == 0) {
-                numOfSeats= (numOfRows / 2) * numOfSeats;
-                int profit = (numOfSeats * 8) + (numOfSeats * 10);
-                String printProfit = "$" + String.valueOf(profit);
-                System.out.println(printProfit);
-            } else {
-                int numOfBackRowSeats = ((int)Math.ceil(numOfRows / 2) + 1) * numOfSeats;
-                int numOfFrontRowSeats = ((int)Math.floor(numOfRows / 2)) * numOfSeats;
-                int profit = (numOfFrontRowSeats * 10) + (numOfBackRowSeats * 8);
-                String printProfit = "$" + String.valueOf(profit);
-                System.out.println(printProfit);
-            }
-        }
-    }
 
-    static void printSeatArrangement(int numOfRows, int numOfSeats) {
+    static void printSeatArrangement(int numOfRows, int numOfSeats, char[][] seatLayout) {
         System.out.println("Cinema:");
         System.out.print("  ");
-        for(int y = 1; y <= numOfSeats; y++) {
-            System.out.print(String.valueOf(y) + " ");
+        for (int y = 0; y <= numOfSeats - 1; y++) {
+            System.out.print(y + 1 + " ");
         }
         System.out.println();
-        for(int i = 1; i <= numOfRows; i++) {
-            System.out.print(String.valueOf(i) + " ");
-            for (int x = 1; x <= numOfSeats; x++) {
-                System.out.print("S ");
+        for (int i = 0; i <= numOfRows - 1; i++) {
+            System.out.print(i + 1  + " ");
+            for (int x = 0; x <= numOfSeats - 1; x++) {
+                System.out.print(seatLayout[i][x] + " ");
             }
             System.out.println();
         }
+        System.out.println();
     }
 
-    static void reserveSeat(int numOfSeats, int numOfRows, int rowNum, int seatNum) {
-        System.out.println("Cinema:");
-        System.out.print("  ");
-        for(int y = 1; y <= numOfSeats; y++) {
-            System.out.print(String.valueOf(y) + " ");
-        }
-        System.out.println();
-        for(int i = 1; i <= numOfRows; i++) {
-            System.out.print(String.valueOf(i) + " ");
-            for (int x = 1; x <= numOfSeats; x++) {
-                if (x == seatNum && i == rowNum) {
-                    System.out.print("B ");
-                } else{
-                    System.out.print("S ");
+    static char[][] reserveSeat (int numOfSeats, int numOfRows, int totalSeats, char[][] seatLayout){
+        int rowNum = rowNum() - 1;
+        int seatNum = seatNum() - 1;
+        ticketPrice(totalSeats, numOfRows, rowNum);
+        for (int i = 0; i <= numOfRows - 1; i++) {
+            for (int x = 0; x < numOfSeats; x++) {
+                if (i == rowNum && x == seatNum) {
+                    seatLayout[rowNum][seatNum] = 'B';
```

Comment:
Why do you have to iterate through this whole list to change one seat? Can you do this without iterating through the list?

Reply:
yes. you are correct. I kind of copy+pasted from my print array and didn't give the logic any thought. Good catch.